### PR TITLE
SDK-192 [Java SDK] Use countries /v2

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic (if any):
- [SDK-192](https://incountry.atlassian.net/browse/SDK-192)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: src/test/java/com/incountry/residence/sdk/http/PopExtractorTest.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        assertTrue(endpoint1.isMidPop());
+
+        //first midpop from the pair with code 'ru'
+        EndPoint endpoint2 = endPointMap.get("ru");
+        assertNotNull(endpoint2);
+        assertEquals("ru", endpoint2.getCountryCode());
+        assertEquals("https://audience.ru.local", endpoint2.getAudience());
+        assertEquals("https://ru.local", endpoint2.getUrl());
+        assertEquals("https://auth.ru.endpoint/oauth2/token", endpoint2.getOauth2Server());
+        assertTrue(endpoint2.isMidPop());
```
Comment (RommeroInCountry) [inline src/test/java/com/incountry/residence/sdk/http/PopExtractorTest.java:82]: what about the case when we have both mini and mid? and mini is first in line? let's add that test case as well

File: src/main/java/com/incountry/residence/sdk/tools/dao/impl/PopExtractor.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            if (endPoint != null && (existingEndPoint == null || !existingEndPoint.isMidPop())) {
+                result.put(transferCountry.getId(), endPoint);
+            }
+        }
+        return result;
+    }
+
+    private static EndPoint extractPop(TransferCountryList.TransferCountry transferCountry) {
+        EndPoint resultEndPoint = null;
+        boolean invalidTransferCountry = isNullOrEmpty(transferCountry.getId()) || !transferCountry.isActive() || transferCountry.getPop() == null || isNullOrEmpty(transferCountry.getOauth2Server());
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/residence/sdk/tools/dao/impl/PopExtractor.java:61]: isInvalid...

File: src/integration/java/com/incountry/residence/sdk/OAuthTest.java [hunk 3]
```diff
@@ -1,113 +0,0 @@
...
-        Record myRecord = new Record("someKey", "someBody");
-
-        StorageNetworkException ex = assertThrows(StorageNetworkException.class, () -> prodStorage.write(countryCode, myRecord));
-        assertEquals(errorMessage, ex.getMessage());
-        List<Class> expectedClasses = Arrays.asList(HttpHostConnectException.class, UnknownHostException.class);
-        Assertions.assertTrue(expectedClasses.contains(ex.getCause().getClass()));
-    }
-
-    @Test
-    void tokenAccessorTest() throws StorageServerException, StorageCryptoException, StorageClientException {
```
Comment (RommeroInCountry) [inline src/integration/java/com/incountry/residence/sdk/OAuthTest.java:90]: this one is still a good and valid test. Let's keep it. We may simply try extracting the first oauth2 server from countries list for this test

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

File: src/main/java/com/incountry/residence/sdk/tools/http/HttpExecutor.java [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 public interface HttpExecutor {
 
-    ApiResponse request(String url, String body, String audience, String region, int retryCount,
+    ApiResponse request(String url, String body, String audience, String oauth2Server, int retryCount,
```
[the remark this answers is not in the document]
Comment (VladimirLevitskiy) [reply src/main/java/com/incountry/residence/sdk/tools/http/HttpExecutor.java:10]: it was from the infra JSON

File: src/test/java/com/incountry/residence/sdk/http/PopExtractorTest.java (hunk 1 above)
Comment (VladimirLevitskiy) [reply src/test/java/com/incountry/residence/sdk/http/PopExtractorTest.java:82]: it's already done at the test 'getCountryListPositiveCountryDuplication'

**Review by robert-par — APPROVED** [review]

**Pull request merged**